### ISSUE TITLE
Firefox Nightly: Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”)

### ISSUE THREAD

**Issue by wellington1993** [emberjs/ember-inspector#1158]
## Issue and Steps to Reproduce

  - Try search on google
  - Get Ember inspector addon error in browser console
  - And can't see google search results

Links:
https://www.google.com/search?newwindow=1&client=firefox-b-d&sxsrf=ALeKk03nSJjNh6tFcy8xukH0BWNDjzapHQ:1584126596986&q=meu.inss.gov.br+login&sa=X&ved=2ahUKEwjKq964k5joAhViK7kGHTyHBF4Q1QIoAXoECAEQAg

## Your environment

Add-on: 3.13.2
Firefox:
Update Channel 	nightly
User Agent 	Mozilla/5.0 (Windows NT 10.0; rv:76.0) Gecko/20100101 Firefox/76.0

## Screenshots
![error-boot-js-ember-inspector-csp](https://user-images.githubusercontent.com/6920053/76653546-c1bee380-6560-11ea-8491-252c74da0c8c.png)

**Comment by CvX** [comment]
CSP can now prevent loading the extension on stable Firefox (118) and on [Safari](https://apps.apple.com/gb/app/emberjs-inspector/id6451499514?mt=12) (17.1)

<img width="1052" alt="image" src="https://github.com/emberjs/ember-inspector/assets/66961/1e54c036-6c8e-4db1-ae6e-d405d4810414">

<img width="873" alt="image" src="https://github.com/emberjs/ember-inspector/assets/66961/34ecd902-f6ad-48fc-8625-b3fbfd13b9cf">

I think the extension has to use [webRequest.onHeadersReceived](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onHeadersReceived) API to modify responses' headers to inject a [CSP nonce](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_inline_script) that would be included in the generated `<script>` elements.

**Comment by RobbieTheWagner** [comment]
I think nightly versions are broken right now from something in manifest v3 not being correct. Can you use the officially released version instead?

**Comment by RobbieTheWagner** [comment]
I think I may have misunderstood, you mean you are on Firefox Nightly, not Ember Inspector Nightly?

**Comment by CvX** [comment]
Yeah, v3 manifest is a different issue (which I also discovered while trying to build the extension today 😃)

CSP problem affects current stable versions of FF and Safari, and all versions of Inspector. (screenshot in my previous comment were taken on the Mozilla Store and App Store versions)

**Comment by RobbieTheWagner** [comment]
Does this problem occur on actual Ember apps? I see the original report mentioned it happened on a google search.

**Comment by patricklx** [comment]
well, its the content-script. It gets injected into all pages to be able to detect ember apps.

**Comment by patricklx** [comment]
I actually see the same in current firefox 
![image](https://github.com/emberjs/ember-inspector/assets/1332320/1dd68725-fa24-48dd-b3bb-ec182b9596c3)

but it loads successfully 
I wonder what the error before is about? `can't access property startTime`

**Comment by patricklx** [comment]
might be related:
https://support.mozilla.org/en-US/questions/1302948

**Comment by CvX** [comment]
> Does this problem occur on actual Ember apps?

Unfortunately it does. See for example https://meta.discourse.org/

But any site that has CSP preventing inline scripts will cause an error, as @patricklx noted.

**Comment by patricklx** [comment]
Can you confirm that this only happens when inspector is installed?

**Comment by patricklx** [comment]
The other error also disappears?

**Comment by patricklx** [comment]
i'm using firefox 118.0.2 and im not seeing that issue.
this only happens with the inspector nightly version

**Comment by patricklx** [comment]
I was wrong.
depending on the sites configuration, it can fail to load ember debug and the boot init scripts.
It can work after a refresh with inspector open, because then ember_debug get injected directly via chrome api instead of an inline script. But it't also broken since boot script still fails and then render tree cannot be obtained
